i3lock.c

```diff
@@ -645,6 +645,120 @@ void handle_screen_resize(void) {
     redraw_screen();
 }
 
+static int read_raw_image_native(uint32_t *dest, FILE *src, int width, int height, int stride) {
+    int count = 0;
+    for (int y = 0; y < height; y++) {
+        size_t n = fread(&dest[y * stride], 1, width * 4, src);
+        count += n;
+        if (n < (size_t)(width * 4))
```

Comment:
Does this code have to worry about integer overflows? (I'll only leave this comment here once, but want to point out that `y * stride` could also overflow and could instead be done via `dest += stride / 4`, which would however "look funny")

Reply:
If we were using size_t I think we should be fine; according to my math, given big enough screens (say, a 64k screen) we would come a bit close to overflowing a 32-bit size_t, but we'd also be coming a bit close to exhausting all of the 32-bit address space given that everything is read into RAM. On a 64-bit system it should be fine.

However, since both y and stride are ints, it can be a 32-bit multiplication on even 64-bit systems. We should probably make sure all the relevant operations are with size_t operands.